doc/howtos/rdtraining/firstui.rst

```diff
@@ -0,0 +1,274 @@
+  We'll see `later <howto/rdtraining/qwebintro>` that there can be other view modes.
+
+Examples can be found everywhere in Odoo, but
+`this one <https://github.com/odoo/odoo/blob/09c59012bf80d2ccbafe21c39e604d6cfda72924/addons/crm/views/crm_lost_reason_views.xml#L57-L70>`__
+is a good example of simple action. It also gives you the structure of an XML data file.
+
+.. exercise:: Add an action.
+
+    Create the ``estate_property_views.xml`` file in the appropriate folder and define it in the
+    ``__manifest__.py`` file.
+
+    Create an action for the model ``estate.property``.
+
+Restart the server, you should see the file loaded in the log.
+
+Menus
+=====
+
+**Reference**: the documentation related to this topic can be found in
+:ref:`reference/data/shortcuts`.
+
+.. note::
+
+    **Goal**: at the end of this section, three menus should be created and the default view is
+    displayed:
+
+    .. image:: firstui/media/estate_menu_root.png
+      :align: center
+      :alt: Root menus
+
+    .. image:: firstui/media/estate_menu_action.png
+      :align: center
+      :alt: First level and action menus
+
+    .. image:: firstui/media/estate_form_default.png
+      :align: center
+      :alt: Default form view
+
+Because menus are somewhat complex to declare there is a ``<menuitem>``
+shortcut to declare an ``ir.ui.menu`` and connect it to the corresponding
+action more easily.
+
+A minimal menu for our ``test_model_action`` would look like:
+
+.. code-block:: xml
+
+    <menuitem id="test_model_menu_action" action="test_model_action"/>
+
+The menu ``test_model_menu_action`` is linked to the action ``test_model_action``, and the action
+is linked to the model ``test.model``. As mentioned before, the action can be seen as the link
+between the menu and the model.
+
+However, menus always follow an architecture; in practice, there are three levels of menus:
+
+1. The root menu, which is displayed in the App switcher
+2. The first level menu, displayed in the top bar
+3. The action menus
+
+   .. image:: firstui/media/menu_01.png
+      :align: center
+      :alt: Root menus
+
+   .. image:: firstui/media/menu_02.png
+      :align: center
+      :alt: First level and action menus
+
+The easiest way to define the structure is to create a structure in the XML file. A minimal
+structure for our ``test_model_action`` would look like:
+
+.. code-block:: xml
+
+    <menuitem id="test_menu_root" name="Test">
+        <menuitem id="test_first_level_menu" name="First Level">
+            <menuitem id="test_model_menu_action" action="test_model_action"/>
+        </menuitem>
+    </menuitem>
+
+.. exercise:: Add the menus.
+
+    Create the ``estate_menus.xml`` file in the appropriate folder and define it in the
+    ``__manifest__.py`` file. Remember the sequential loading of the data files ;-)
+
+    Create the three levels of menus for the ``test_model_action`` action.
```

Comment:
Create the three levels of menus (as it's shown in the goal) for the ... Maybe say something to remember that your menus should look like the ones in the goal ?